src/components/lib/Tooltip.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef, useState } from "react"
+import classNames from "classnames"
+import PropTypes from "prop-types"
+
+const Tooltip = ({
+  triggerElement = <button>Help</button>,
+  children,
+  revealOnHover = true,
+  className,
+}) => {
+  const [showContent, setShowContent] = useState(false)
+  const toggleContent = () => {
+    setShowContent(!showContent)
+  }
+
+  const tooltipRef = useRef(null)
+  useEffect(() => {
+    const clickOutside = (e) => {
+      if (tooltipRef.current && !tooltipRef.current.contains(e.target)) {
+        setShowContent(false)
+      }
+    }
+
+    // Only add the listener if the dropdown is open, and make sure to clean
+    // it up once it's closed
+    if (showContent) {
+      document.addEventListener("click", clickOutside)
+    } else {
+      document.removeEventListener("click", clickOutside)
```

Comment:
Given that `clickOutside` is a different instance every time the `useEffect` function body runs, will this work? Wondering if it would be cleaner, instead of the `else`, to have `clickOutside` clean up the event listener itself (by adding the `document.removeEventListener` call  inside that function)...

Reply:
It works (tested with console.log), but I agree it would be cleaner to just put it inside the event listener itself. :)

Reply:
Lol stuff like that makes me realize I still don't understand the fullness of hook voodoo. Or of event listener management... As the case may be 😅

Reply:
😆 It's a little confusing generally. 🧹 Cleaned up in 938b637!